docs/reference/sparql-reference.md

```diff
@@ -0,0 +1,280 @@
+prefix dce: <http://purl.org/dc/elements/1.1/>
+prefix dc: <http://purl.org/dc/terms/>
+
+# selecting only unique instances of the three variables 
+SELECT DISTINCT ?term ?property ?value WHERE 
+{
+  # order of the variables in the triple
+	?term ?property ?value .
+    # the variable property is an annotation property
+  	?property a owl:AnnotationProperty .
+  # selects entities that are native to ontology (in this case MONDO)
+	FILTER (isIRI(?term) && regex(str(?term), "^http://purl.obolibrary.org/obo/MONDO_"))
+    # removes triples where the variable value is blank
+  	FILTER(!isBlank(?value))
+  # listing the allowed annotation properties
+  FILTER (?property NOT IN (dce:creator, dce:date, IAO:0000115, IAO:0000231, IAO:0100001, mondo:excluded_subClassOf, mondo:excluded_from_qc_check, mondo:excluded_synonym, mondo:pathogenesis, mondo:related, mondo:confidence, dc:conformsTo, mondo:should_conform_to, oboInOwl:consider, oboInOwl:created_by, oboInOwl:creation_date, oboInOwl:hasAlternativeId, oboInOwl:hasBroadSynonym, oboInOwl:hasDbXref, oboInOwl:hasExactSynonym, oboInOwl:hasNarrowSynonym, oboInOwl:hasRelatedSynonym, oboInOwl:id, oboInOwl:inSubset, owl:deprecated, rdfs:comment, rdfs:isDefinedBy, rdfs:label, rdfs:seeAlso, RO:0002161, skos:broadMatch, skos:closeMatch, skos:exactMatch, skos:narrowMatch))
+}
+```
+
+### Checking for misused replaced_by 
+adaptable for checking that a property is used in a certain way
+
+```SPARQL
+
+# adding prefixes used
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+
+# selecting only unique instances of the three variables 
+SELECT DISTINCT ?entity ?property ?value WHERE {
+ # the variable property is IAO_0100001 (item replaced by)
+ VALUES ?property { <http://purl.obolibrary.org/obo/IAO_0100001> }
+ 
+ # order of the variables in the triple
+ ?entity ?property ?value .
+ # removing entities that have either owl:deprecated true or oboInOwl:ObsoleteClass (these entities are the only ones that should have replaced_by)
+ FILTER NOT EXISTS { ?entity owl:deprecated true }
+ FILTER (?entity != oboInOwl:ObsoleteClass)
+}
+# arrange report by entity variable
+ORDER BY ?entity
+```
+
+## Count
+
+### Count class by prefixes
+
+```SPARQL
+# this query counts the number of classes you have with each prefix (eg number of MONDO terms, CL terms, etc.)
+
+# adding prefixes used
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix obo: <http://purl.obolibrary.org/obo/>
+
+# selecting 2 variables, prefix and numberOfClasses, where number of classes is a count of distinct cls
+SELECT ?prefix (COUNT(DISTINCT ?cls) AS ?numberOfClasses) WHERE 
+{
+  # the variable cls is a class 
+  ?cls a owl:Class .
+  # removes any cases where the variable cls is blank
+  FILTER (!isBlank(?cls))
+  # Binds the variable prefix as the prefix of the class (eg. MONDO, CL, etc.)
+  BIND( STRBEFORE(STRAFTER(str(?cls),"http://purl.obolibrary.org/obo/"), "_") AS ?prefix)
```

Comment:
you should add a FILTER for ?cls that STRSTARTS with the OBO purl blurp

Reply:
not quite sure what this means?

Reply:
oh i get it, actually, I quite like that there is a blank to tell me that they are not obo purls